### PR TITLE
delete source by uuid

### PR DESCRIPTION
**Pull request by hitarthpattani**
### Description (*)

User can delete any source with UUID

### Fixed Issues (if relevant)

1. magento/async-import#85: Delete Source API

### Manual testing scenarios (*)

URL (DELETE request): [DOMAIN]/rest/V1/import/source/7e620a88-6a70-4011-a822-2369e29e6a2f
Headers: Authorization: Bearer <admin_user_token>

**Expected Result:**
bool (`true`) or exception with message `Source with uuid \"%1\" does not exist`.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by hitarthpattani** [comment]
Hello @nuzil 

I have implemented delete operation using provided guidelines. 

Please check and do the needful.

File: app/code/Magento/ImportService/Model/SourceRepository.php [hunk 1]
```diff
@@ -115,13 +124,10 @@ public function delete(SourceInterface $source)
 
     /**
      * @inheritdoc
-     *
-     * @throws CouldNotDeleteException
-     * @throws NoSuchEntityException
      */
-    public function deleteByUuid($uuid)
+    public function deleteByUuid(string $uuid): bool
```
Comment (nuzil) [inline app/code/Magento/ImportService/Model/SourceRepository.php]: Please change implementation, that return of operation is void and not a boolean.
So on success we have to receive []

File: app/code/Magento/ImportService/Model/Source/Command/DeleteByUuidInterface.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * @see \Magento\ImportService\Api\SourceRepositoryInterface
+ */
+interface DeleteByUuidInterface
+{
+    /**
+     * Delete the source by uuid. If source is not found do nothing
+     *
+     * @param string $uuid
+     * @return bool
+     * @throws CouldNotDeleteException
```
Comment (nuzil) [inline app/code/Magento/ImportService/Model/Source/Command/DeleteByUuidInterface.php]: We do not throw CouldNotDeleteException, please delete

File: app/code/Magento/ImportService/Api/SourceRepositoryInterface.php [hunk 3]
```diff
@@ -47,10 +47,12 @@ public function getList(\Magento\Framework\Api\SearchCriteriaInterface $criteria
     public function delete(\Magento\ImportService\Api\Data\SourceInterface $source);
 
     /**
-     * Deletes source by UUID
+     * Delete the source by uuid. If source is not found, NoSuchEntityException is thrown
      *
      * @param string $uuid
      * @return bool
+     * @throws \Magento\Framework\Exception\CouldNotDeleteException
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/SourceRepositoryInterface.php]: We do not throw CouldNotDeleteException, please delete

File: app/code/Magento/ImportService/Model/SourceRepository.php (hunk 1 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Model/SourceRepository.php]: Hello, @nuzil I have changed the return type bool to void.

**Pull request merged**